app/controllers/posts_controller.rb

```diff
@@ -106,24 +108,28 @@ def new
 
   def create
     import_thread and return if params[:button_import].present?
-    preview and return if params[:button_preview].present?
 
     @post = current_user.posts.new(post_params)
+    @post.written = @post.replies.new(written_params)
+    @post.written.user = current_user
+
+    preview and return if params[:button_preview].present?
     @post.settings = process_tags(Setting, :post, :setting_ids)
     @post.content_warnings = process_tags(ContentWarning, :post, :content_warning_ids)
     @post.labels = process_tags(Label, :post, :label_ids)
 
-    unless @post.save
+    begin
+      @post.save!
+      flash[:success] = "You have successfully posted."
```

Comment:
I'd leave this at the end; minimizing stuff in a begin/rescue is good practice unless we need to.